tests/OtherTest.php

```diff
@@ -582,4 +582,63 @@ function() {
 		$this->expectWarningMessage( '@type Not_Supported_Type is not supported by Parse.ly. Please use a type mentioned in https://www.parse.ly/help/integration/jsonld#distinguishing-between-posts-and-pages' );
 		self::$parsely->construct_parsely_metadata( $options, $post_obj );
 	}
+
+	/**
+	 * Test that test_display_admin_warning action returns a warning when there is no key
+	 *
+	 * @covers \Parsely::display_admin_warning
+	 */
+	public function test_display_admin_warning_without_key() {
+		$options           = get_option( \Parsely::OPTIONS_KEY );
+		$options['apikey'] = '';
+		$this->set_options( $options );
+
+		ob_start();
+		self::$parsely->display_admin_warning();
+		$output = trim( ob_get_contents() );
+		ob_end_clean();
```

Comment:
If we're having to capture output to see if it contained a string or not, then that's a code smell that we really should be testing against a boolean function that handles the logic of whether it should display or not i.e. a new `should_display_admin_warning()`. 

Then we can have a unit test to test the various conditions of that, and an optional second one that tests the contents of the message (that's probably redundant in this case though as you'd be testing a hard-coded string against a hard-coded expected string).